### ISSUE TITLE
First word shows as double points until refresh

### ISSUE THREAD

**Issue by pkolano** [cdot/Xanado#49]
On the latest, after the first player went, it showed they played twice with double the score.  It fixed itself after refresh, but my brother said this has happened before on the first turn.  Also, the red dots are still showing up and kind of randomly moving between players as the game progresses.  It used to be pretty stable up until the last few updates.

![doublescore](https://user-images.githubusercontent.com/20344426/176820141-bac006c2-d0b1-49e5-9852-b8514186f467.png)

**Comment by cdot** [comment]
I can't reproduce this, though I can guess how it might happen. Once the UI is initially loaded, subsequent updates are incremental, by sending messages over a socket. One of the following must be happening:
* the server is sending the same message twice. In this case I would expect to see that in the server turn history. A refresh (which replays the history) eliminates the duplication, so this is unlikely. Did the other players also see this symptom, or was it just you?
* there are two events arriving for the same message. This would imply an error in the WebSockets code. I think this is highly unlikely in a mainstream browser like Chrome, though I suppose it is possible..
* there is more than one handler for the same event. I suspect this is the most likely explanation, but can only be confirmed by looking at the browser console log.
It would help to know what code revision is being run.

**Comment by cdot** [comment]
The server seems to be behaving itself, as evidenced by the refresh correcting errors. It uses WebSocket, which in turn uses TCP, so messages should be lossless and ordered, even in the face of a flaky network. I think it's fair to assume that messages are arriving in the client in order, and without duplication.

There are a number of scenarios in which it might be possible for a socket event to end up with multiple handlers attached to it. I've added some asserts to detect this case; I would very much like to see the browser console if any of these asserts fail. I also added a sequence number to detect when an event is handled twice. Let me know if that cures it, and if it doesn't, I really need a snapshot of the console log from one of the faulty UIs.
